### PR TITLE
[Kernel] Remove marlin moe templating on thread_m_blocks

### PR DESCRIPTION
**Pull request by tlrmchlsmth**
Reduce Marlin MoE binary size and compile time by removing the template parameter `thread_m_blocks` in `MarlinMoE`. On current main, the `MarlinMoE` fn is completely ignoring `thread_m_blocks`, so deleting the template parameter is a free win.

**This PR:**
`Wheel dist/vllm-0.6.1.post2+cu124-cp38-abi3-linux_x86_64.whl is within the allowed size (150.02 MB).`

**Current main:**
`Wheel dist/vllm-0.6.1.post2+cu124-cp38-abi3-linux_x86_64.whl is within the allowed size (195.64 MB).`

### CODE REVIEW

**Comment by tlrmchlsmth** [comment]
We should get @LucasWilkinson added as a coauthor before merging.

**Review by simon-mo — APPROVED** [review]

**Review by mgoin — APPROVED** [review]

File: csrc/moe/marlin_moe_ops.cu [hunk 1]
```diff
@@ -1872,7 +1849,6 @@ void marlin_mm_moe_f16i4(const void* A, const void* B, void* C,
     for (int m_block = 0; m_block < tot_m_blocks;
          m_block += 4 * exec_cfg.max_m_blocks) {
       // make it max possible value
```
Comment (mgoin) [inline csrc/moe/marlin_moe_ops.cu:1851]: Should we remove the comment as well?

Comment (tlrmchlsmth) [reply csrc/moe/marlin_moe_ops.cu:1851]: Actually, I was a bit overzealous here -- restoring this so we can bring the unsupported shapes error message back to its former glory

File: csrc/moe/marlin_moe_ops.cu [hunk 2]
```diff
@@ -1711,31 +1703,16 @@ exec_config_t determine_thread_config(int prob_m, int prob_n, int prob_k,
...
-                                                                       \
-  __CALL_IF_MOE(W_TYPE, 4, N_BLOCKS, K_BLOCKS, false, -1, NUM_THREADS) \
-  __CALL_IF_MOE(W_TYPE, 4, N_BLOCKS, K_BLOCKS, false, 2, NUM_THREADS)  \
-  __CALL_IF_MOE(W_TYPE, 4, N_BLOCKS, K_BLOCKS, false, 4, NUM_THREADS)  \
-  __CALL_IF_MOE(W_TYPE, 4, N_BLOCKS, K_BLOCKS, false, 8, NUM_THREADS)
+#define CALL_IF_MOE(W_TYPE, N_BLOCKS, K_BLOCKS, NUM_THREADS)        \
+  __CALL_IF_MOE(W_TYPE, N_BLOCKS, K_BLOCKS, true, 0, NUM_THREADS)   \
+  __CALL_IF_MOE(W_TYPE, N_BLOCKS, K_BLOCKS, true, 0, NUM_THREADS)   \
+  __CALL_IF_MOE(W_TYPE, N_BLOCKS, K_BLOCKS, true, 0, NUM_THREADS)   \
+  __CALL_IF_MOE(W_TYPE, N_BLOCKS, K_BLOCKS, true, 0, NUM_THREADS)   \
```
Comment (ElizaWszola) [inline csrc/moe/marlin_moe_ops.cu:1710]: Shouldn't all of this be just one `__CALL_IF_MOE`?

**Pull request merged**